api/src/main/java/org/openmrs/module/mirebalais/require/RequireUtil.java

```diff
@@ -35,6 +36,15 @@ public static String sessionLocationDoesNotHaveTag(LocationTagDescriptor descrip
         return new String("typeof sessionLocation !== 'undefined' && !hasMemberWithProperty(sessionLocation.tags, 'display','" + descriptor.name() + "')");
     };
 
+    // note that Java 8 Nashorn script engine support does not (fully?) support ECMAScript6 so we can't use arrow notation or map, etc
+    public static String visitHasEncounterOfType(EncounterTypeDescriptor descriptor) {
+        return new String("typeof visit !== 'undefined' && typeof visit.encounters !== 'undefined' && hasMemberThatEvaluatesTrue(visit.encounters, (function(encounter) { return encounter.encounterType.uuid === '" + descriptor.uuid() + "' } ))");
```

Comment:
[Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is ES5

Reply:
```suggestion
        return new String("visit && visit.encounters && " + 
          "visit.encounters.some(function(encounter) { " +
          "  return encounter.encounterType.uuid === '" + descriptor.uuid() +
          "' })");
```